### PR TITLE
OCP-34940 - test vsphere credential changes gets reflected

### PR DESCRIPTION
**Pull request by miyadav**
Here are the logs - 
    And the output should match:                                                       # features/step_definitions/common.rb:33
      | SuccessfulAttachVolume |
waiting for operation up to 3600 seconds..
      # Restoring original state
    When I run the :delete admin command with:                                         # features/step_definitions/cli.rb:31
      [18:17:00] INFO> Shell Commands: oc delete pod mypod --config=/root/workdir/e92b52530f58-/ocp4_admin.kubeconfig
      pod "mypod" deleted
      
      [18:17:02] INFO> Exit Status: 0
      | object_type       | pod   |
      | object_name_or_id | mypod |
    Then the step should succeed                                                       # features/step_definitions/common.rb:4
waiting for operation up to 3600 seconds..
    When I run the :delete client command with:                                        # features/step_definitions/cli.rb:13
      [18:17:02] INFO> Shell Commands: oc delete pvc pvc --config=/root/workdir/e92b52530f58-/ocp4_admin.kubeconfig
      persistentvolumeclaim "pvc" deleted
      
      [18:17:04] INFO> Exit Status: 0
      | object_type       | pvc |
      | object_name_or_id | pvc |
    Then the step should succeed                                                       # features/step_definitions/common.rb:4
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
      [18:17:04] INFO> === After Scenario: PVCs can still be provisioned after the password has been changed vSphere ===
      [18:17:04] INFO> Shell Commands: rm -r -f -- /root/workdir/e92b52530f58-
      
      [18:17:04] INFO> Exit Status: 0
      [18:17:04] INFO> === End After Scenario: PVCs can still be provisioned after the password has been changed vSphere ===

1 scenario (1 passed)
26 steps (26 passed)
4m30.857s
[18:17:04] INFO> === At Exit ===

Job Url - https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/176516/console
Related Bug - https://bugzilla.redhat.com/show_bug.cgi?id=1821280

### CODE REVIEW

File: features/machine/machine_misc.feature [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+   Given I use the "openshift-config" project
+   When as admin I successfully merge patch resource "cm/cloud-provider-config" with:
+     |  {"data": {"immutable": "false"}} |
+
+   Then I use the "openshift-machine-api" project
+   When 240 seconds have passed
+
+   Then I obtain test data file "cloud/misc/PVC.yaml"
+   And I run oc create over "PVC.yaml" replacing paths:
+     | n      | openshift-machine-api       |
```
Comment (jhou1) [inline features/machine/machine_misc.feature:33]: Please format the table.

File: features/machine/machine_misc.feature [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+   Then the step should succeed
+
+   Given I obtain test data file "cloud/misc/nginx-pod.yaml"
+   When I run oc create over "nginx-pod.yaml" replacing paths:
+     | n      | openshift-machine-api       |
```
Comment (jhou1) [inline features/machine/machine_misc.feature:38]: Please format the table.

File: features/machine/machine_misc.feature [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+
+   # Restoring original state
+   When I run the :delete admin command with:
+      | object_type       | pod   |
+      | object_name_or_id | mypod |
+    Then the step should succeed
+
+   When I run the :delete client command with:
+      | object_type | pvc |
+      | object_name_or_id | pvc |
```
Comment (jhou1) [inline features/machine/machine_misc.feature:62]: Please format the table.

Comment (miyadav) [reply features/machine/machine_misc.feature:62]: done

File: features/machine/machine_misc.feature (hunk 2 above)
Comment (miyadav) [reply features/machine/machine_misc.feature]: done

File: features/machine/machine_misc.feature (hunk 1 above)
Comment (miyadav) [reply features/machine/machine_misc.feature]: done

File: features/machine/machine_misc.feature [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+     |  {"data": {"immutable": "false"}} |
+
+   Then I use the "openshift-machine-api" project
+   When 240 seconds have passed
+
+   Then I obtain test data file "cloud/misc/PVC.yaml"
+   And I run oc create over "PVC.yaml" replacing paths:
+     | n | openshift-machine-api | 
+   Then the step should succeed
+
```
Comment (sunzhaohua2) [inline features/machine/machine_misc.feature:35]: And admin ensures "pvc" pvc is deleted after scenario

File: features/machine/machine_misc.feature [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+   Given I obtain test data file "cloud/misc/nginx-pod.yaml"
+   When I run oc create over "nginx-pod.yaml" replacing paths:
+     | n | openshift-machine-api |
+
```
Comment (sunzhaohua2) [inline features/machine/machine_misc.feature:39]: And admin ensures "mypod" pod is deleted after scenario

File: features/machine/machine_misc.feature [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+   # Restoring original state
+   When I run the :delete admin command with:
+      | object_type       | pod   |
+      | object_name_or_id | mypod |
+    Then the step should succeed
+
+   When I run the :delete client command with:
+      | object_type       | pvc |
+      | object_name_or_id | pvc |
```
Comment (sunzhaohua2) [inline features/machine/machine_misc.feature:62]: remove these lines.

File: features/machine/machine_misc.feature (hunk 4 above)
Comment (miyadav) [reply features/machine/machine_misc.feature:35]: Thanks added this

File: features/machine/machine_misc.feature (hunk 5 above)
Comment (miyadav) [reply features/machine/machine_misc.feature:39]: Thanks added this

File: features/machine/machine_misc.feature (hunk 6 above)
Comment (miyadav) [reply features/machine/machine_misc.feature]: done

File: features/machine/machine_misc.feature [hunk 7]
```diff
@@ -0,0 +1,56 @@
...
+      | force |                              |
+   And the step should succeed
+ 
+   Then I get project events
+   And the output should match:
+     | SuccessfulAttachVolume |
+
+   # Restoring original state
+   And admin ensures "mypod" pod is deleted after scenario
+   Then admin ensures "pvc" pvc is deleted after scenario
```
Comment (jhou1) [inline features/machine/machine_misc.feature:56]: The `admin ensures delete` step should follow closely to its create step, please move these lines to the correct position.

File: testdata/cloud/misc/vsphere-creds_test.yaml [hunk 8]
```diff
@@ -0,0 +1,16 @@
+apiVersion: v1
+data:
+  vcsa2-qe.vmware.devcluster.openshift.com.password: CWdycjRiQFhqpjA2
+  vcsa2-qe.vmware.devcluster.openshift.com.username: QSRtbW4pc1RyYXRvckBlMmUdvG9jZWw=
```
Comment (jhou1) [inline testdata/cloud/misc/vsphere-creds_test.yaml:4]: Are they random passwords?(just in case)

File: features/machine/machine_misc.feature (hunk 7 above)
Comment (miyadav) [reply features/machine/machine_misc.feature]: thanks ..did that

File: testdata/cloud/misc/vsphere-creds_test.yaml (hunk 8 above)
Comment (miyadav) [reply testdata/cloud/misc/vsphere-creds_test.yaml:4]: yes ..

File: features/machine/machine_misc.feature [hunk 9]
```diff
@@ -0,0 +1,55 @@
...
+
+   Then I get project events
+   And the output should match:
+     | ServerFaultCode: Cannot complete login due to an incorrect user name or password |
+
+   Then I run the :replace admin command with:
+      | _tool | oc                           |
+      | f     | vsphere-creds_original.yaml  |
+      | force |                              |
+   And the step should succeed
```
Comment (akostadinov) [inline features/machine/machine_misc.feature:50]: This needs to be a clean-up step in case something fails.

Comment (miyadav) [reply features/machine/machine_misc.feature:50]: Thanks , important step , will need help on this

**Comment by miyadav** [comment]
/unhold

Here are the results 
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/181789/console

File: features/step_definitions/secret_restore.rb [hunk 10]
```diff
@@ -0,0 +1,13 @@
+Given /^admin ensures the changed secret "([^"]*)" is restored in "([^"]*)" project after scenario$/ do | secret , project |
```
Comment (akostadinov) [inline features/step_definitions/secret_restore.rb:1]: No need to add a new step. We already have a step `/^the #{QUOTED} (\w+) is recreated( by admin)? in the#{OPT_QUOTED} project after scenario$/`. Somehting like:
```
Given the "vsphere-creds" secret is recreated by admin in the "kube-system" project after scenario
```

Comment (miyadav) [reply features/step_definitions/secret_restore.rb]: Thank you , reused and validated , looks good 
https://mastern-jenkins-csb-openshift-qe.cloud.paas.psi.redhat.com/job/Runner-v3/183113/console

**Comment by akostadinov** [comment]
Thank you!

**Pull request merged**